src/fheroes2/dialog/dialog_selectfile.cpp

```diff
@@ -478,7 +478,8 @@ namespace
                     needRedraw = true;
                 }
                 else if ( !listboxEvent && le.isAnyKeyPressed()
-                          && ( !isTextLimit || fheroes2::Key::KEY_BACKSPACE == le.getPressedKeyValue() || fheroes2::Key::KEY_DELETE == le.getPressedKeyValue() )
+                          && ( filename.size() >= lengthLimit || fheroes2::Key::KEY_BACKSPACE == le.getPressedKeyValue()
```

Comment:
Hi,@ihhub, I'm not able to input any character in save game dialog.
Here we should check that there is no input limit: `filename` is lower than `lengthLimit `:
```suggestion
                          && ( filename.size() < lengthLimit || fheroes2::Key::KEY_BACKSPACE == le.getPressedKeyValue()
```

Reply:
Hmm... no idea why I did this...